bundles/org.openhab.binding.bambulab/src/main/java/org/openhab/binding/bambulab/internal/PrinterHandler.java

```diff
@@ -0,0 +1,309 @@
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import pl.grzeslowski.jbambuapi.PrinterClient;
+import pl.grzeslowski.jbambuapi.PrinterClientConfig;
+import pl.grzeslowski.jbambuapi.PrinterWatcher;
+import pl.grzeslowski.jbambuapi.Report;
+
+/**
+ * The {@link PrinterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class PrinterHandler extends BaseThingHandler implements PrinterWatcher.StateSubscriber {
+    private static final Pattern DBM_PATTERN = Pattern.compile("^(-?\\d+)dBm$");
+    private Logger logger = LoggerFactory.getLogger(PrinterHandler.class);
+
+    private @Nullable PrinterClient client;
+
+    public PrinterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!LED_CHAMBER_LIGHT_CHANNEL.equals(channelUID.getId())
+                && LED_WORK_LIGHT_CHANNEL.equals(channelUID.getId())) {
+            return;
+        }
+        var ledNode = LED_CHAMBER_LIGHT_CHANNEL.equals(channelUID.getId()) ? CHAMBER_LIGHT : WORK_LIGHT;
+        var bambuCommand = "ON".equals(command.toFullString()) ? on(ledNode) : off(ledNode);
+        sendCommand(bambuCommand);
+    }
+
+    @Override
+    public void initialize() {
+        var config = getConfigAs(PrinterConfiguration.class);
+
+        if (config.serial.isEmpty()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "@token/printer.handler.init.noSerial");
+            return;
+        }
+        logger = LoggerFactory.getLogger(PrinterHandler.class.getName() + "." + config.serial);
+
+        if (config.hostname.isEmpty()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "@token/printer.handler.init.noHostname");
+            return;
+        }
+
+        var scheme = config.scheme;
+        var port = config.port;
+        var rawUri = "%s%s:%d".formatted(scheme, config.hostname, port);
+        URI uri;
+        try {
+            uri = new URI(rawUri);
+        } catch (URISyntaxException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                    "@token/printer.handler.init.invalidHostname[\"" + rawUri + "\"]");
+            return;
+        }
+
+        if (config.accessCode.isEmpty()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "@token/printer.handler.init.noAccessCode");
+            return;
+        }
+
+        if (config.username.isEmpty()) {
+            config.username = PrinterClientConfig.LOCAL_USERNAME;
+        }
+
+        updateStatus(UNKNOWN);
+
+        PrinterClient localClient;
+        try {
+            localClient = client = new PrinterClient(
```

Comment:
Is this blocking? As there is a seperate `connect` method i suppose not, double checking as this method should return really fast and therefore it should not  wait for request/response or other blocking calls.

Reply:
yes, in scheduler I'm running connect methods. This one just creates the object